### DISCUSSION TITLE
Tree-Based Polymetric View

### DISCUSSION THREAD

**Discussion by mircealungu** [git-truck/git-truck#92] [General]
Folks,

Do you think you will get to investigate the possibility of a view like this besides the circle-packing layout? 
    
<img width="1089" alt="image" src="https://user-images.githubusercontent.com/464519/154487186-f85435fa-a0d3-40ef-bed8-b645cf1c9ed8.png">

The benefits would be that your tool is not a one-trick-pony, but can be argued to be a way of visualizing a system in multiple ways.

If you don't think you can do it, I have other students who are looking for research projects, and I could propose it to them. And then you can focus on showing more info in the circle packing layout (e.g. recent changes, churn, and maybe an interactive commit picker, where one can go back in history to see how the project looked at that time). 

Cheers,
M. 

P.S. Apropos, it seems this is a good intro to d3 if anybody is interested: https://wattenberger.com/blog/react-and-d3

**Comment by nimrossum** [comment]
D3 has support for a view like this. I would argue that it would take little time to implement it

We can support any [d3-hiearchy type](https://github.com/d3/d3-hierarchy#api-reference)

- [ ] Hierarchy (Stratify)
- [ ] Cluster
- [ ] Tree
- [x] Treemap (Treemap Tiling)
- [ ] Partition
- [x] Pack

We currently have implemented support for Treemap and Pack (Circle packing)

**Reply by nimrossum** [reply]
I just experimented with the Tree for a few minutes:
![image](https://user-images.githubusercontent.com/1959615/158403292-6d188039-f8aa-4c2c-9d39-c6804fef4f3d.png)

**Reply by mircealungu** [reply]
a few more minutes and you're there :)

**Reply by nimrossum** [reply]
![image](https://user-images.githubusercontent.com/1959615/158408595-a690f673-4649-4193-9e84-f5010d379dac.png)

whoops

**Reply by mircealungu** [reply]
In the summer we should recreate some of these views in high resolution and prepare a software visualization exhibition :)

**Reply by nimrossum** [reply]
![image](https://user-images.githubusercontent.com/1959615/158409312-56628fea-a652-41cc-ac02-039fe21e52d3.png)

**Reply by nimrossum** [reply]
I pushed these changes to a branch. I don't know if we will actually implement it. Until it becomes polymetric, it is not very useful.

**Reply by nimrossum** [reply]
The cluster one looks interesting, it seems that all leaves are at the bottom:
![image](https://user-images.githubusercontent.com/1959615/158412482-82271b02-48a7-4b2e-999d-d548e232baa6.png)

**Reply by nimrossum** [reply]
![image](https://user-images.githubusercontent.com/1959615/158413548-83502752-a730-4902-9a6b-fc376a3d1ffa.png)
Radail looks crazy, I think @hojelse looked into something like this previously.

**Reply by nimrossum** [reply]
This is similar:
https://observablehq.com/@observablehq/why-visualization-helps-developers
